static/monorail-2.0.yaml

```diff
@@ -650,6 +670,162 @@ paths:
           schema:
             $ref: '#/definitions/Error'
 
+  /hooks:
+    get:
+      description: Get all hooks
+      operationId: hooksGetAll
+      responses:
+        200:
+          description: Successfully retrieved the hook configuration
+          schema:
+            type: object
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+      summary: Get all hooks
+      tags:
+      - /api/2.0
+      x-authentication-type:
+      - jwt
+      x-privileges:
+      - Read
+      - hooksRead
+      x-view: hook.2.0.json
+    post:
+      consumes:
+      - application/json
+      description: Create a new hook.
+      operationId: hooksPost
+      parameters:
+      - description: configuration hook to be created
+        in: body
+        name: body
+        required: true
+        schema:
+            $ref: '#/definitions/hooks.2.0_HookPost'
+      responses:
+        201:
+          description: Successfully created new hook
+          schema:
+            type: object
+        409:
+          description: Duplicated hook.
+          schema:
+            $ref: '#/definitions/Error'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+      summary: Post a hook
+      tags:
+      - /api/2.0
+      x-authentication-type:
+      - jwt
+      x-privileges:
+      - Write
+      - hooksWrite
+      x-swagger-schema: hooks.2.0.json#/definitions/HookPost
+      x-view: hook.2.0.json
+    x-swagger-router-controller: hooks
+  /hooks/{identifier}:
+    get:
+      description: Get hook by id
+      operationId: hooksGetById
+      parameters:
+      - description: id of the hook you wish to delete
+        in: path
+        name: identifier
+        required: true
+        type: string
+      responses:
+        200:
+          description: Successfully retrieved the specified hook
+          schema:
+            type: object
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+      summary: Get all hooks
+      tags:
+      - /api/2.0
+      x-authentication-type:
+      - jwt
+      x-privileges:
+      - Read
+      - hooksRead
```

Comment:
@pengz1 @iceiilin `x-view` is needed for any route that returns a hook object in the response body.